openmc/mgxs/mdgxs.py

```diff
@@ -605,20 +655,45 @@ def print_xs(self, subdomains='all', nuclides='all', xs_type='macro'):
-                                              xs_type=xs_type, value='rel_err',
-                                              delayed_groups=[delayed_group])
-                        average = average.flatten()[0]
-                        rel_err = rel_err.flatten()[0] * 100.
-                        string += '{:.2e} +/- {:1.2e}%'.format(average, rel_err)
-                        string += '\n'
+                    average_xs = self.get_xs(nuclides=[nuclide],
+                                             subdomains=[subdomain],
+                                             xs_type=xs_type, value='mean',
+                                             delayed_groups=[delayed_group])
+                    rel_err_xs = self.get_xs(nuclides=[nuclide],
+                                             subdomains=[subdomain],
+                                             xs_type=xs_type, value='rel_err',
+                                             delayed_groups=[delayed_group])
+                    rel_err_xs = rel_err_xs * 100.
+
+                    if self.num_polar > 1 or self.num_azimuthal > 1:
+                        # Loop over polar, azimuthal, and energy group ranges
+                        for pol in range(len(polar_bins) - 1):
+                            pol_low, pol_high = polar_bins[pol: pol + 2]
+                            for azi in range(len(azimuthal_bins) - 1):
+                                azi_low, azi_high = azimuthal_bins[azi: azi + 2]
+                                string += '\t\tPolar Angle: [{0:5f} - {1:5f}]'.format(
+                                    pol_low, pol_high) + \
+                                    '\tAzimuthal Angle: [{0:5f} - {1:5f}]'.format(
+                                    azi_low, azi_high) + '\n'
+                                for group in range(1, self.num_groups + 1):
+                                    bounds = \
+                                        self.energy_groups.get_group_bounds(group)
+                                    string += '\t' + template.format('', group,
+                                                                     bounds[0],
+                                                                     bounds[1])
+                                    string += '{1:.2e} +/- {1:.2e}%'.format(
+                                        average_xs[pol, azi, group - 1],
+                                        rel_err_xs[pol, azi, group - 1])
+                                    string += '\n'
+                                string += '\n'
+                    else:
+                        # Loop over energy groups ranges
+                        for group in range(1, self.num_groups+1):
+                            bounds = self.energy_groups.get_group_bounds(group)
+                            string += template.format('', group, bounds[0], bounds[1])
+                            string += '{1:.2e} +/- {1:.2e}%'.format(
```

Comment:
This is actually correct.  `self.energy_groups.get_group_bounds` gets the right energy boundaries corresponding to `group` and `average_xs` (and `rel_err_xs`) are already ordered in standard group ordering.

As an example, here is the output of 2-group total xs call to `print_xs` in water:
```	Subdomain      =	(1, 1, 25)
	Cross Sections [cm^-1]:
            Group 1 [0.625      - 20000000.0eV]:	6.46e-01 +/- 7.17e-02
            Group 2 [0.0        - 0.625     eV]:	1.98e+00 +/- 1.65e-01
```

As you can see the group numbers are correctly related to the energy bins and the thermal x/s is larger than fast, as expected.